app/src/main/java/com/blink22/blink22_boilerplate/base/BaseActivity.java

```diff
@@ -0,0 +1,78 @@
+package com.blink22.blink22_boilerplate.base;
+
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+
+import com.blink22.blink22_boilerplate.AppInstance;
+import com.blink22.blink22_boilerplate.ApplicationComponent;
+import com.blink22.blink22_boilerplate.R;
+import com.blink22.blink22_boilerplate.base.loading.LoadingFragmentDialog;
+import com.blink22.core.base.BaseView;
+import com.tapadoo.alerter.Alerter;
+import com.tapadoo.alerter.OnHideAlertListener;
+
+public abstract class BaseActivity extends AppCompatActivity implements BaseView{
```

Comment:
@mhemdan  I think we can use [`IcePick`](https://github.com/frankiesardo/icepick/) in the base activity and all the extending ones won't suffer from having the state parameters nulled once the activity destroyed. Check the library [Readme file](https://github.com/frankiesardo/icepick/blame/master/README.md#L21)